crates/sequencer/src/sequencer.rs

```diff
@@ -629,53 +630,77 @@ where
                 .map_err(|e| anyhow!(e))?;
             let (notify, rx) = oneshot_channel();
             let request = BlobWithNotifier { blob, notify };
-            inscription_queue
+            self.da_service
+                .get_send_transaction_queue()
                 .send(request)
                 .map_err(|_| anyhow!("Bitcoin service already stopped!"))?;
-            let tx_id = rx
-                .await
-                .map_err(|_| anyhow!("DA service is dead!"))?
-                .map_err(|_| anyhow!("Send transaction cannot fail"))?;
-
-            self.ledger_db
-                .set_last_sequencer_commitment_l2_height(BatchNumber(
-                    commitment_info.l2_height_range.end().0,
-                ))
-                .map_err(|_| {
-                    anyhow!("Sequencer: Failed to set last sequencer commitment L2 height")
-                })?;
-
-            debug!("Commitment info: {:?}", commitment_info);
-
-            let l2_start = l2_range_to_submit.start().0 as u32;
-            let l2_end = l2_range_to_submit.end().0 as u32;
-            if let Some(db_config) = self.config.db_config.clone() {
-                match PostgresConnector::new(db_config).await {
-                    Ok(pg_connector) => {
-                        pg_connector
-                            .insert_sequencer_commitment(
-                                Into::<[u8; 32]>::into(tx_id).to_vec(),
-                                l2_start,
-                                l2_end,
-                                commitment.merkle_root.to_vec(),
-                                CommitmentStatus::Mempool,
-                            )
-                            .await
-                            .map_err(|_| {
-                                anyhow!("Sequencer: Failed to insert sequencer commitment")
-                            })?;
-                    }
-                    Err(e) => {
-                        warn!("Failed to connect to postgres: {:?}", e);
-                    }
-                }
-            }
 
-            // Clear state diff.
+            info!("Sent commitment to DA queue");
+
+            // TODO: this causes state diff to be lost on restart/error, fix that
             self.ledger_db.set_state_diff(vec![])?;
             self.last_state_diff = vec![];
 
-            info!("New commitment. L2 range: #{}-{}", l2_start, l2_end,);
+            // Store this only in memory to indicate there is an in-progress
+            // commitment process happenning
+            self.last_committed_l2_height = Some(*commitment_info.l2_height_range.end());
```

Comment:
This does not indicate that there is an in-progress commitment but rather sets the value of the "already committed". That's how we interpret this.

What happens if this value was set but submitting the commitment failed inside the async block below? The sequencer would fall out of sync when looking at the value in this var vs what is stored inside ledger_db.

Assuming it goes through, there's no need to introduce `self.last_committed_l2_height` member variable since `get_commitment_info` will use the ID that we've just written to ledger db.

In short, i do believe we can make this much better.

Reply:
The thing is, we need something that indicates that there is an asynchronous commit happening right now and check based on that. Otherwise it is gonna commit every time a new block is produced once the commitment conditions are satisfied.

But we also don't want it to block the further commitments incoming if the previous one didn't finish yet, hence we can't just use a simple boolean to indicate that. We can rename this to `last_commit_try_l2_height`, and reset this back to its old value if the previous one failed, but that is still problematic. Because if second commitment arrived and pushed to queue, right after that first commitment failed, and second commitment succeded. This scenario will cause a gap for the l2 height range of the first commitment.

Maybe we can requeue the commitment if it failed with the hopes that everything is gonna turn out fine?

Just putting my though flow out there. LMK what you think.

Reply:
Or for now, maybe the best course of action is to block the commitment submission if there is one going on right now, even though its not optimal.

Reply:
> The thing is, we need something that indicates that there is an asynchronous commit happening right now and check based on that. Otherwise it is gonna commit every time a new block is produced once the commitment conditions are satisfied.

that's fine?

Reply:
wait how is that fine? lets say the min soft confirmation limit is 100, and we reached the block 100 and got ourselves a 1-100 commitment. when 101 comes, it will submit another commitment for 1-101, 102 will submit 1-102 etc.

Reply:
ah, you mean that.

yeah, that's actually not fine.

what about another ledger db method that returns a vec of pending "commitment ranges" so that we can check that on commitment controller.

then in the next prs we can build on that, for restarts etc.

Reply:
I see your point.

We need a way to lock specific commitments to prevent running multiple commitments txs for the same range.

How about you add a HashSet for in-progress commitments where the entry is the l2 end range and just check if that one has been submitted already then you skip. If it wasn't, add and then spawn.
Once the spawned task is done, remove it from the list. This should serve as some kind of a locking mechanism.

Reply:
> I see your point.
> 
> We need a way to lock specific commitments to prevent running multiple commitments txs for the same range.
> 
> How about you add a HashSet for in-progress commitments where the entry is the l2 end range and just check if that one has been submitted already then you skip. If it wasn't, add and then spawn. Once the spawned task is done, remove it from the list. This should serve as some kind of a locking mechanism.

We definitely need to store the range as you said. But if we only store it in memory, it is going to be problematic to handle error cases, as it could leave gaps in between commits. So I believe it is best to store it in a db as Esad suggested above like "in_progress_commitment_ranges". And maybe now, or at least in the future, that would allow us to compare that table with the finalized commitments and figure out what to do from there on.